generate/genhcl/genhcl.go

```diff
@@ -123,6 +136,41 @@ func (h HCL) String() string {
 		h.Label(), h.Condition(), h.Body(), h.Range().HostPath())
 }
 
+// Header returns the HCL header based on the comment style.
+func Header(comment CommentStyle) string {
+	return stdfmt.Sprintf("%s "+HeaderMagic+"\n\n", comment)
+}
+
+// DefaultHeader returns the header for the default comment style.
+func DefaultHeader() string {
+	return Header(DefaultComment)
+}
+
+// CommentStyleFromString returns the comment style given an string.
+func CommentStyleFromString(str string) CommentStyle {
+	switch str {
+	case "//":
+		return SlashComment
+	case "#":
+		return HashComment
+	default:
+		panic(errors.E(errors.ErrInternal, "invalid comment style"))
+	}
+}
+
+// CommentStyleFromConfig returns the CommentStyle from the configuration or the
+// default if not defined.
+func CommentStyleFromConfig(root *config.Root) CommentStyle {
```

Comment:
Ah got it, I had forgotten about that requirement, it makes sense to fetch the config in this function then.
However I suggest changing the type of the function arg because we don't need the complete `*config.Root` here:

```suggestion
func CommentStyleFromConfig(tmConfig *hcl.Terramate) CommentStyle {
```